utils/plugins/utils.go

```diff
@@ -60,33 +40,49 @@ func CheckPluginsVersionAndConvertIfNeeded() (err error) {
 		return
 	}
 
-	plugins, err := readPluginsConfig()
-	if err != nil {
-		return
-	}
-	if plugins.Version != coreutils.GetPluginsConfigVersion() {
-		err = fmt.Errorf("expected plugins version in 'plugins.yaml is %d but the actual value is %d", coreutils.GetPluginsConfigVersion(), plugins.Version)
-	}
-	return
+	return readPluginsConfigAndConvertV0tToV1IfNeeded()
 }
 
-func readPluginsConfig() (*PluginsV1, error) {
-	plugins := new(PluginsV1)
+func readPluginsConfigAndConvertV0tToV1IfNeeded() (err error) {
 	content, err := getPluginsConfigFileContent()
 	if err != nil {
```

Comment:
Can be simplified by checking:
```suggestion
        // Return without converting in case of an error, or if the plugins.yml file already exists, which indicates a conversion has already been made.
	if err != nil || len(content)!=0 {
```
Then removing the second if.